src/components/ScreenFlatList/index.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { FlatList } from '..';
+import Header from './header';
+import { ScreenFlatListProps } from './types';
+
+const ScreenFlatList = ({
+    title,
+    itemComponent,
+    headerComponent: HeaderComponent,
+    subHeaderComponent: SubHeaderComponent,
+    dataSource,
+}: ScreenFlatListProps) => {
+    const header = HeaderComponent ? <HeaderComponent title={title} /> : <Header title={title} />;
+    const subHeader = SubHeaderComponent ? <SubHeaderComponent title={title} /> : null;
```

Comment:
here we need in the SubHeaderComponent pass all data returned by the dataSource, then we will add a header component for the flat list that will be the subHeaderComponent here, then in the flat list internally pass the dataSource response